ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh

```diff
@@ -281,12 +282,12 @@ aws --region $REGION cloudformation create-stack --stack-name ${stack_name} \
 wait "$!"
 echo "Created stack"
 
+echo "$stack_name" > "${SHARED_DIR}/bastion_host_stack_name"
```

Comment:
I wonder why we need this change in this step?

Reply:
The UPI reboot case requires a bastion host, but no such step in the current [UPI workflow](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/rehearse/aws/upi/provision/cucushift-installer-rehearse-aws-upi-provision-chain.yaml#L7), all resources, including VPC, were created under by only one step `upi-install-aws-cluster`.
So I [added a bastion host](https://github.com/openshift/release/pull/52689/files#diff-13d91d14b7d179ffee59edb4b21315b2e665f13da600f72e67ffb800fc015f73R8-R9) after cluster created.

`${SHARED_DIR}/bastion_host_stack_name` is used by [UPI deprovision](https://github.com/openshift/release/pull/52689/files#diff-73cd47f7713a8b00880513b47cdb7a2d51b8c4f75dd13d793a7fda58be2f1df1R31-R40), we need to make sure the bastion host is removed before deleting [UPI stacks](https://github.com/openshift/release/pull/52689/files#diff-73cd47f7713a8b00880513b47cdb7a2d51b8c4f75dd13d793a7fda58be2f1df1R54-R57) (including VPC)

Reply:
In this step, bastion ignition s3 bucket, and stack name is recorded on line 63 and 270, are those duplicated ?

Reply:
`to_be_removed_*` are for common use, but new ones are for UPI  process only, especially it's a condition that check if bastion host were created in this workflow.

Reply:
or using this way?
```
```

Reply:
Based on we are injecting `aws-provision-bastionhost` step in `cucushift-installer-rehearse-aws-upi-provision` chain, I just checked the code, if we are going to inject `aws-deprovision-stacks` and `aws-deprovision-s3buckets` into `cucushift-installer-rehearse-aws-upi-provision` , it would also work, right?

Reply:
Like this ?
```
chain:
  as: cucushift-installer-rehearse-aws-upi-deprovision
  steps:
  - chain: gather
  - ref: gather-aws-console
  - ref: aws-deprovision-stacks             < - 1. delete bastion
  - ref: aws-deprovision-s3buckets          < - 2. delete bastion s3
  - ref: upi-deprovision-aws                < - 3. delete cluster stacks
  - ref: ipi-deprovision-deprovision
  - ref: upi-deprovision-aws-check-destroy
  documentation: |-
    Destroy the AWS UPI cluster.
```

Reply:
Yes, does that work?